kratos/includes/global_pointer.h

```diff
@@ -22,272 +22,409 @@
 #include "includes/serializer.h"
 #include "includes/key_hash.h"
 
-namespace Kratos {
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
 
-template<class TDataType>
-class GlobalPointer {
-private:
+///@name Kratos Classes
+///@{
 
-  /// Pointer to the data
-  TDataType * mDataPointer;
+/**
+* @class GlobalPointer
+* @ingroup KratosCore
+* @brief This class is a wrapper for a pointer to a data that is located in a different rank
+* @details This class is a wrapper for a pointer to a data that is located in a different rank
+* @tparam TDataType The type of the data pointed by the GlobalPointer
+* @see GlobalPointerHasher
+* @author Carlos A. Roig
+*/
+template<class TDataType>
+class GlobalPointer
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Definition of element type
+    using element_type = TDataType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+   /**
+    * @brief Default constructor
+    * @details This should never be called as we need a local pointer to exists
+    */
+    GlobalPointer() {
+        mDataPointer = nullptr;
 #ifdef KRATOS_USING_MPI
-  int mRank;
+        this->mRank = 0;
```

Comment:
why do you want to initialize it? if it is not a correct value better leave it undefined no?

Reply:
The diff is terrible, the code was already like that , see line 48 in previous implementation